hack/kind-up.sh

```diff
@@ -167,6 +172,88 @@ check_registry_cache_availability() {
   done
 }
 
+# The default StorageClass which comes with `kind' is configured to use
+# rancher.io/local-path (see [1]) provisioner, which defaults to `hostPath'
+# volume (see [2]).  However, `hostPath' does not expose any metrics via
+# kubelet, while `local' (see [3]) does. On the other hand `kind' does not
+# expose any mechanism for configuring the StorageClass it comes with (see [4]).
+#
+# This function annotates the default StorageClass with `defaultVolumeType: local',
+# so that we can later scrape the various `kubelet_volume_stats_*' metrics
+# exposed by kubelet (see [5]).
+#
+# References:
+#
+# [1]: https://github.com/rancher/local-path-provisioner
+# [2]: https://kubernetes.io/docs/concepts/storage/volumes/#hostpath
+# [3]: https://kubernetes.io/docs/concepts/storage/volumes/#local
+# [4]: https://github.com/kubernetes-sigs/kind/blob/main/pkg/cluster/internal/create/actions/installstorage/storage.go
+# [5]: https://kubernetes.io/docs/reference/instrumentation/metrics/
+_setup_kind_sc_default_volume_type() {
+    echo "Configuring default StorageClass for kind cluster ..."
+    kubectl get storageclasses standard -o yaml | \
+        yq '.metadata.annotations.defaultVolumeType = "local"' | \
+        kubectl apply -f -
+}
+
+# The rancher.io/local-path provisioner at the moment does not support volume
+# resizing (see [1]). There is an open PR, which is scheduled for the next
+# release around May, 2024 (see [2]). Until [2] is merged we will use a custom
+# local-path provisioner with support for volume resizing.
+#
+# This function should be called after setting up the containerd registries on
+# the kind nodes.
+#
+# References:
+#
+# [1]: https://github.com/rancher/local-path-provisioner
+# [2]: https://github.com/rancher/local-path-provisioner/pull/350
+_setup_kind_with_lpp_resize_support() {
```

Comment:
You could add a TODO to remind us to remove these hacks when when the corresponding local-path-provisioner release is adopted.